fearless/Modules/NewWallet/SendFlow/ChooseRecipient/ChooseRecipientViewController.swift

```diff
@@ -0,0 +1,196 @@
+import UIKit
+import SoraUI
+
+final class ChooseRecipientViewController: UIViewController, ViewHolder {
+    typealias RootViewType = ChooseRecipientViewLayout
+
+    let presenter: ChooseRecipientPresenterProtocol
+
+    private var tableViewModel: ChooseRecipientTableViewModel?
+    private var locale = Locale.current
+    private var isFirstLayoutCompleted: Bool = false
+
+    private lazy var searchActivityIndicatory: UIActivityIndicatorView = .init(style: .white)
+
+    init(presenter: ChooseRecipientPresenterProtocol) {
+        self.presenter = presenter
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func loadView() {
+        view = ChooseRecipientViewLayout()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setupKeyboardHandler()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        clearKeyboardHandler()
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        isFirstLayoutCompleted = true
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        presenter.setup()
+
+        configure()
+
+        navigationController?.setNavigationBarHidden(true, animated: false)
+    }
+
+    func configure() {
+        rootView.searchView.textField.delegate = self
+
+        rootView.tableView.registerClassForCell(SearchPeopleTableCell.self)
+
+        rootView.tableView.tableFooterView = UIView()
+
+        rootView.tableView.dataSource = self
+        rootView.tableView.delegate = self
+
+        rootView.navigationBar.backButton.addTarget(self, action: #selector(backButtonClicked), for: .touchUpInside)
+        rootView.nextButton.addTarget(self, action: #selector(nextButtonClicked), for: .touchUpInside)
+        rootView.scanButton.addTarget(self, action: #selector(scanButtonClicked), for: .touchUpInside)
+        rootView.historyButton.addTarget(self, action: #selector(historyButtonClicked), for: .touchUpInside)
+        rootView.pasteButton.addTarget(self, action: #selector(pasteButtonClicked), for: .touchUpInside)
+    }
+
+    @objc private func backButtonClicked() {
+        presenter.didTapBackButton()
+    }
+
+    @objc private func scanButtonClicked() {
+        presenter.didTapScanButton()
+    }
+
+    @objc private func pasteButtonClicked() {
+        rootView.searchView.textField.text = UIPasteboard.general.string
```

Comment:
we can add notifications for uipasteboard and if UIPastboard receive valid address show Paste button and hide them if address is not valid. Feature for discussion